src/treelearner/feature_histogram.hpp

```diff
@@ -135,6 +158,10 @@ class FeatureHistogram {
     int best_threshold = -1;
     int best_dir = 1;
 
+    if (constraints != nullptr) {
+      constraints->InitializeIndices(1);
+    }
```

Comment:
I don't think we need constraints in this function, you can pass it, but don't need to call it.